src/signaling/backend_configuration.go

```diff
@@ -148,6 +148,101 @@ func NewBackendConfiguration(config *goconf.ConfigFile) (*BackendConfiguration,
 	}, nil
 }
 
+func (b *BackendConfiguration) RemoveBackend(host string) {
+	delete(b.backends, host)
+}
+
+func (b *BackendConfiguration) UpsertHost(host string, backends []*Backend) {
+	existingIndex := 0
+	for _, existingBackend := range b.backends[host] {
+		found := false
+		index := 0
+		for _, newBackend := range backends {
+			if existingBackend == newBackend {
+				found = true
+				backends = append(backends[:index], backends[index+1:]...)
+				continue
+			}
+			index++
+		}
+		if !found {
+			b.backends[host] = append(b.backends[host][:existingIndex], b.backends[host][existingIndex+1:]...)
+		}
+		existingIndex++
+	}
+
+	b.backends[host] = append(b.backends[host], backends...)
+}
+
+func (b *BackendConfiguration) Reload(config *goconf.ConfigFile) {
+	type configuredBackend struct {
+	}
+	if backendIds, _ := config.GetString("backend", "backends"); backendIds != "" {
+		seenIds := make(map[string]struct{})
+		seenHosts := make(map[string]map[string]struct{})
+		for _, id := range strings.Split(backendIds, ",") {
+			id = strings.TrimSpace(id)
+			if id == "" {
+				continue
+			}
+
+			if _, ok := seenIds[id]; !ok {
+				seenIds[id] = struct{}{}
+			}
+
+			u, _ := config.GetString(id, "url")
+			if u == "" {
+				log.Printf("Backend %s is missing or incomplete, skipping", id)
+				continue
+			}
+
+			if u[len(u)-1] != '/' {
+				u += "/"
+			}
+			parsed, err := url.Parse(u)
+			if err != nil {
+				log.Printf("Backend %s has an invalid url %s configured (%s), skipping", id, u, err)
+				continue
+			}
+
+			if _, ok := seenHosts[parsed.Host]; !ok {
```

Comment:
This will only save the first id for a host. Also you could directly parse the `secret` and construct the `Backend` object here instead of parsing again below.